common/src/Microsoft.Azure.IIoT.Core/src/Extensions/UriEx.cs

```diff
@@ -90,6 +90,10 @@ public static class UriEx {
                 fileDevice = localPath.Substring(0, index + 5);
                 pathAndQuery = localPath.Substring(index + 5);
             }
+            else if ((index = localPath.IndexOf("sock", StringComparison.InvariantCultureIgnoreCase)) != -1) {
```

Comment:
better here to define strings (e.g. ".sock")as consts and derive the index modifier from constants

Reply:
the "socket syntax is in linux is ".sock" and in windows is "/sock", therefore the "sock" check is enough for both, not that strict nevertheless...